src/Domain.LinnApps/ImportBooks/ImportBookService.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace Linn.Stores.Domain.LinnApps.ImportBooks
+{
+    using System.Collections.Generic;
+    using Linn.Common.Persistence;
+    using Linn.Stores.Domain.LinnApps.Parts;
+
+    public class ImportBookService : IImportBookService
+    {
+        private readonly IRepository<ImportBookInvoiceDetail, ImportBookInvoiceDetailKey> InvoiceDetailRepository;
+        private readonly IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> OrderDetailRepository;
+        private readonly IRepository<ImportBookPostEntry, ImportBookPostEntryKey> PostEntryRepository;
+
+
+        public ImportBookService(
+            IRepository<ImportBookInvoiceDetail, ImportBookInvoiceDetailKey> invoiceDetailRepository,
+            IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> orderDetailRepository,
+            IRepository<ImportBookPostEntry, ImportBookPostEntryKey> postEntryRepository)
+        {
+            this.InvoiceDetailRepository = invoiceDetailRepository;
+            this.OrderDetailRepository = orderDetailRepository;
+            this.PostEntryRepository = postEntryRepository;
+        }
+
+        public void Update(
+            ImportBook from,
+            IEnumerable<ImportBookInvoiceDetail> invoiceDetails,
+            IEnumerable<ImportBookOrderDetail> orderDetails,
+            IEnumerable<ImportBookPostEntry> postEntries)
+        {
+            this.UpdateInvoiceDetails(invoiceDetails);
+
+            this.UpdateOrderDetails(orderDetails);
+
+            this.UpdatePostEntries(postEntries);
+        }
+
+        public void UpdateInvoiceDetails(IEnumerable<ImportBookInvoiceDetail> details)
+        {
+            foreach (var detail in details)
+            {
+                var currentDetail = this.InvoiceDetailRepository.FindById(new ImportBookInvoiceDetailKey(detail.ImportBookId, detail.LineNumber));
+
+                if (currentDetail == null)
+                {
+                    this.CreateInvoiceDetail(detail);
+                }
+                else
+                {
+                    currentDetail.InvoiceNumber = detail.InvoiceNumber;
```

Comment:
Will this update what's in the repo by reference? Or would I have to do it much higher up/use an actual ImportBookInvoiceDetailsFacadeService update method with the sort of from & to parameters passed into the domain service

Reply:
Getting an object via its repository GetById(<id>) and then updating it should persist any changes after Commit() is called (which is done at the end of the FacadeService's Update(). Add(). etc implementations). 

Unless you put an .AsNoTracking() call in your GetById() method that is... that would tell the dbContext not to track updates to entities. Been caught out by that a few times

Reply:
Okay coool, should be all good here then. Thanks